src/Types/ID.php

```diff
@@ -35,6 +36,14 @@ public function val(): bool|float|int|string|object
 
     public function __toString(): string
     {
-        return (string) $this->value;
+        if (is_bool($this->value)) {
+            return $this->value === true ? '1' : '0';
+        }
+
+        if (is_scalar($this->value)) {
+            return (string) $this->value;
+        }
+
+        return (string) $this->value->__toString();
```

Comment:
`__toString()` should always return a string.  We should remove the explicit type cast.

Reply:
done